pubsub/natsstreaming/natsstreaming_test.go

```diff
@@ -0,0 +1,518 @@
+			subscriptionType:       "topic",
+			deliverAll:             "true",
+		}
+		fakeMetaData := pubsub.Metadata{
+			Properties: fakeProperties,
+		}
+		m, err := parseNATSStreamingMetadata(fakeMetaData)
+
+		assert.NoError(t, err)
+
+		assert.NotEmpty(t, m.natsURL)
+		assert.NotEmpty(t, m.natsStreamingClusterID)
+		assert.NotEmpty(t, m.subscriptionType)
+		assert.NotEmpty(t, m.natsQueueGroupName)
+		assert.NotEmpty(t, m.deliverAll)
+
+		assert.Equal(t, fakeProperties[natsURL], m.natsURL)
+		assert.Equal(t, fakeProperties[natsStreamingClusterID], m.natsStreamingClusterID)
+		assert.Equal(t, fakeProperties[subscriptionType], m.subscriptionType)
+		assert.Equal(t, fakeProperties[consumerID], m.natsQueueGroupName)
+		assert.Equal(t, fakeProperties[deliverAll], m.deliverAll)
+	})
+	t.Run("using deliverNew", func(t *testing.T) {
+		fakeProperties := map[string]string{
+			natsURL:                "nats://foo.bar:4222",
+			natsStreamingClusterID: "testcluster",
+			consumerID:             "consumer1",
+			subscriptionType:       "topic",
+			deliverNew:             "true",
+		}
+		fakeMetaData := pubsub.Metadata{
+			Properties: fakeProperties,
+		}
+		m, err := parseNATSStreamingMetadata(fakeMetaData)
+
+		assert.NoError(t, err)
+
+		assert.NotEmpty(t, m.natsURL)
+		assert.NotEmpty(t, m.natsStreamingClusterID)
+		assert.NotEmpty(t, m.subscriptionType)
+		assert.NotEmpty(t, m.natsQueueGroupName)
+		assert.NotEmpty(t, m.deliverNew)
+
+		assert.Equal(t, fakeProperties[natsURL], m.natsURL)
+		assert.Equal(t, fakeProperties[natsStreamingClusterID], m.natsStreamingClusterID)
+		assert.Equal(t, fakeProperties[subscriptionType], m.subscriptionType)
+		assert.Equal(t, fakeProperties[consumerID], m.natsQueueGroupName)
+		assert.Equal(t, fakeProperties[deliverNew], m.deliverNew)
+	})
+
+	t.Run("using startAtTimeDelta", func(t *testing.T) {
+		fakeProperties := map[string]string{
+			natsURL:                "nats://foo.bar:4222",
+			natsStreamingClusterID: "testcluster",
+			consumerID:             "consumer1",
+			subscriptionType:       "topic",
+			startAtTimeDelta:       "1h",
+		}
+		fakeMetaData := pubsub.Metadata{
+			Properties: fakeProperties,
+		}
+		m, err := parseNATSStreamingMetadata(fakeMetaData)
+
+		assert.NoError(t, err)
+
+		assert.NotEmpty(t, m.natsURL)
+		assert.NotEmpty(t, m.natsStreamingClusterID)
+		assert.NotEmpty(t, m.subscriptionType)
+		assert.NotEmpty(t, m.natsQueueGroupName)
+		assert.NotEmpty(t, m.startAtTimeDelta)
+
+		assert.Equal(t, fakeProperties[natsURL], m.natsURL)
+		assert.Equal(t, fakeProperties[natsStreamingClusterID], m.natsStreamingClusterID)
+		assert.Equal(t, fakeProperties[subscriptionType], m.subscriptionType)
+		assert.Equal(t, fakeProperties[consumerID], m.natsQueueGroupName)
+		dur, _ := time.ParseDuration(fakeProperties[startAtTimeDelta])
+		assert.Equal(t, m.startAtTimeDelta, dur)
+	})
+	//t.Run("using startAtTime and startAtTimeFormat", func(t *testing.T) {})
+}
+
+func TestSubscriptionOptions(t *testing.T) {
```

Comment:
Same here. Would you please use table driven test pattern?

Reply:
grouped sub-tests as table driven tests